src/shim/index.js

```diff
@@ -122,6 +122,18 @@ const commands = {
 function read() {
     let line = system.stdin.readLine();
     if (line) {
+        if (line === 'NOOP') {
```

Comment:
I want to understand this more. If node send `NOOP` and then phantomjs doesn't return an ACK then you are saying that it can fill backlog. I think I get what you are saying. Would it make more sense to `clearInterval` on node side instead of checking with a boolean. Because right now, it will call `hearbeat` every `100ms`. Instead, it might be more performant to wait until the ack is returned to set `setInterval` again (or `setTimeout`)?